src/tools/web/kdb.js

```diff
@@ -0,0 +1,116 @@
+// kdb.js - small library to access Elektra's kdb via node.js
+const { exec } = require('child_process')
+
+// constants
+const ERR_KEY_NOT_FOUND = 'Did not find key'
+
+// KDBError
+function KDBError (message) {
+    this.name = 'KDBError'
+    this.message = message || ''
+}
+KDBError.prototype = Error.prototype
+
+// remove newline from the end of a string
+const trimNewline = (str) =>
+  str.substring(0, str.length - 1)
+
+// execute a script while catching and parsing errors
+const safeExec = (script) => new Promise((resolve, reject) =>
+  exec(script, (err, stdout, stderr) => {
+    if (err) {
+      const errors = err.message.split('\n')
+      // ignore error if it's "key not found"
+      if (!(errors.length > 1 && errors[1] === ERR_KEY_NOT_FOUND)) {
+        return reject(err)
+      }
+    }
+    if (stderr) {
+      if (stderr !== ERR_KEY_NOT_FOUND) {
+        return reject(new KDBError(stderr))
+      } else {
+        return resolve() // no key found, return no value
+      }
+    }
+    const result = trimNewline(stdout)
+    if (!result) {
+      return resolve() // empty result, return no value
+    }
+    return resolve(result)
+  })
+)
+
+// escape strings by surrounding them with ""
+const escapeValues = (template, ...values) =>
+  template.reduce((acc, part, i) => {
+    let val = values[i - 1]
+    if (typeof val === 'string') val = `"${val}"`
```

Comment:
I'm no security expert but I think it is problematic, that strings that get passed to clusterd and elektrad make it to the commandline.
e.g.: I can do that `http GET 'localhost:1234/kdb/user/$(touch $HOME/somefile)'`, which creates a file in my home directory.

At least the values should be escaped with single instead of double quotes here.